File: Python/Programmers/Level 2/H-Index.py
# h-index => 역순 정렬 후 max(f(n) >= i)
# 즉, f(n)이 i랑 같거나 작아지는 순간이 h-index가 됨

def solution(citations):
    citations.sort(reverse=True) # h-index를 구하기 위해 역순 정렬
    for i in range(len(citations)):
        if citations[i] <= i: # 공식 적용 즉, citations[i] 값이 i랑 같거나 작아지는 순간이 h-index
            return i
          
    return len(citations) # 만약 citations[i] 값이 i랑 같거나 작아지는 순간이 없다면 citations 길이가 h-index가 됨

File: JavaScript/Programmers/Level 2/H-Index.js
// H-Index는 주어진 논문들의 인용횟수를 내림차순 한 후 인덱스 값이 해당 요소 값 이상이면 해당 인덱스 값이 h-index가 됨

function solution(citations) {
    citations.sort((a, b) => b - a); // 내림차순 정렬
    for (let i = 0; i < citations.length; i++) {
        if (i >= citations[i]) return i; // 인덱스 값이 해당 요소의 값 이상이면 인덱스 return
    }
    return citations.length; // 인덱스 값이 큰 경우가 없다면 citations의 길이 return
}
